### PR TITLE
feat(book): An online book documentation for presenterm using `mdbook`

### PR DESCRIPTION
**Pull request by pwnwriter**
Hello @mfontanini,

There's so much happening that I'd like to rearrange some things to make the process smoother.

- I've included a book that utilizes [mdBook][mdbook] with the [catppuccin theme][catppuccin_theme] alongside the default ones.
  **Mdbook ci will run on every new push on master branch and update your docs automatically to new branch called `book`**

 Deployed here: [Presenterm Netlify Deployment][netlify_deploy]

 ***I tried using GitHub Pages, but it wasn't rendering assets.***

GitHub Pages deploy: [Presenterm GitHub Pages Deployment][github_pages_deploy]

So, I suggest sticking with Netlify.

- Refactored entry to prevent panics (#124)
- Added dependabot to make pr on deps updates 
  - It'll  be easy for you to manage deps as i see you were bumping them manually. 

Feel free to remove or add anything; I'm open to whatever changes you think are suitable. :v: 

[mdbook]: https://github.com/rust-lang/mdBook
[catppuccin_theme]: https://github.com/catppuccino/mdbook
[netlify_deploy]: https://presenterm.netlify.app
[github_pages_deploy]: https://pwnwriter.github.io/presenterm

### CODE REVIEW

**Comment by pwnwriter** [comment]
Also i'm sorry, I should have checked it before

screenshot_2024-01-10_11-40-39

You don't need to provide v0.1.2 or You can proceed with just the version
eg: 
```bash
./releaser.sh 0.1.2 # or soo
```

Seems like [aliesbelik/presenterm](https://github.com/aliesbelik/presenterm) is making a pr . i'm letting you guys make changes !

File: docs/src/guides/installation.md [hunk 1]
```diff
@@ -51,3 +51,19 @@ nix run github:mfontanini/presenterm  # to run from github repo
...
+
+## Arch linux repository (Aur)
+
+Presenterm is also available in the aur. You can use any AUR helper to install.
+
+#### Building from git  
+> ***recommended for upstream updates***
+
+```bash
+paru/yay -S presenterm-git
```
Comment (mfontanini) [inline docs/src/guides/installation.md:63]: I removed the PGKBUILD that were here before, do these still work without that?

**Review by mfontanini — COMMENTED** [review]
This is awesome, thanks for setting it up! I left a few comments; let me know if you'd like me to take care of them, I have some free time these days.

Regarding deployment, this needs to go to github pages IMO. I don't want to have to handle setting up anything that I can avoid. Images are probably not showing because of relative paths (I left a comment on that).

**Comment by pwnwriter** [comment]
Yes ! I got everything you said , I'll make changes and resolve tomorrow morning! Gotta sleep rn.

**Comment by mfontanini** [comment]
Another one: the links in the root README file in the repo point to various places in the docs, e.g. in the ["features" section](https://github.com/mfontanini/presenterm?tab=readme-ov-file#features). Those should now point to `/docs/src`. Once this is deployed we can make them point to the mdbook but until then they should point somewhere valid.

File: docs/src/assets/demo-image.png
[the remark this answers is not in the document]
Comment (pwnwriter) [reply docs/src/assets/demo-image.png:1]: I copied just to conflict with your current readme. We can move after deployment.

File: docs/src/about.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# About
+
+Welcome to the [Presenterm][github] documentation, A reliable resource
```
[the remark this answers is not in the document]
Comment (pwnwriter) [reply docs/src/about.md:3]: Oh? okay :)

**Comment by pwnwriter** [comment]
> Another one: the links in the root README file in the repo point to various places in the docs, e.g. in the ["features" section](https://github.com/mfontanini/presenterm?tab=readme-ov-file#features). Those should now point to /docs/src. Once this is deployed we can make them point to the mdbook but until then they should point somewhere valid.

 I think we can streamline the home README by including only essential information and providing a direct link to the documentation?.

File: docs/src/about.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+Markdown format and easily render them within your terminal interface.
+
+## Quick demo
+This is how demo presentation looks like: 
+![demo]
+
+
+<!-- Links -->
+[github]: https://github.com/mfontanini/presenterm/
+[demo]: /assets/demo.gif
```
[the remark this answers is not in the document]
Comment (pwnwriter) [reply docs/src/about.md]: Oh yes, it works now

https://pwnwriter.github.io/presenterm 

Deployed from gh-pages branch.

File: docs/src/SUMMARY.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+[About](./about.md)
+
+# Guides
+- [installation](./guides/installation.md)
+- [Introduction](./guides/introduction.md)
+- [Themes](./guides/themes.md)
+- [Layout](./guides/layout.md)
+- [Code highlighting](./guides/code-highlight.md)
+- [Pdf export](./guides/pdf-export.md)
```
[the remark this answers is not in the document]
Comment (pwnwriter) [reply docs/src/SUMMARY.md]: Sure

File: .github/dependabot.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+version: 2
```
[the remark this answers is not in the document]
Comment (pwnwriter) [reply .github/dependabot.yml:1]: Okay

**Comment by mfontanini** [comment]
I like having the list of features showing up near the top given there's various tools that serve the same purpose but I think presenterm has more features than the rest of them so that's the big differentiator.

File: .github/workflows/mdbook-build.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+name: MdBook Build and Deploy
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
Comment (mfontanini) [inline .github/workflows/mdbook-build.yml:7]: `pull_request` shouldn't be here right? You only want to deploy this on a merge.

File: docs/src/assets/demo-image.png
[the remark this answers is not in the document]
Comment (mfontanini) [reply docs/src/assets/demo-image.png:1]: Can't you also change the readme's image paths? I've tried to be careful not to add images larger than possible as they make the git history unnecessarily heavier size-wise so this isn't ideal.

File: .github/workflows/mdbook-build.yml (hunk 6 above)
Comment (pwnwriter) [reply .github/workflows/mdbook-build.yml:7]: If someone makes a PR for docs, it will be easy for you to go on checking live site.

Comment (mfontanini) [reply .github/workflows/mdbook-build.yml:7]: Where exactly would the mdbook be deployed in that case? Wouldn't that go straight into mfontanini.github.io/presenterm?

Comment (pwnwriter) [reply .github/workflows/mdbook-build.yml:7]: It should be their own fork. But they need to setup gh pages for that. 

but nvm, you can remove it.

**Comment by mfontanini** [comment]
Merging this against an intermediate branch. Will make changes and merge to master once it's ready.

**Comment by mfontanini** [comment]
Thanks @pwnwriter !

**Pull request merged**